src/widgets/panel/panel.js

```diff
@@ -131,21 +131,23 @@ export default function panel({
 
       return {
         ...widget,
-        dispose() {
+        dispose(...args) {
           unmountComponentAtNode(getContainerNode(container));
 
           if (typeof widget.dispose === 'function') {
-            widget.dispose();
+            widget.dispose.call(this, ...args);
```

Comment:
We can add a test on the `panel` to be sure that we don't lose the context rather than only rely on stories.

Reply:
I tried but I had difficulties with the `rewire` system, it makes it complicated.